### PR TITLE
Rules feedback

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- translate rules example to yaml
- add deprecation warnings for old rule-like policies

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wochinge** [comment]
@ricwo I get a lot warnings regarding `unsatisfied checkpoints` when training on rules data in yaml format. Any idea? I used the `examples/rules`

**Comment by wochinge** [comment]
@Ghostvv Do you know these warnings?
Screenshot 2020-07-09 at 18 48 03

File: rasa/core/training/generator.py [hunk 1]
```diff
@@ -727,6 +727,9 @@ def _issue_unused_checkpoint_notification(
         collected_start = set()
         collected_end = set()
         for step in self.story_graph.story_steps:
+            if step.is_rule:
```
Comment (wochinge) [inline rasa/core/training/generator.py:730]: fix most of the warnings with this hack 👀

**Comment by Ghostvv** [comment]
the error is the first warning, it thinks that all your stories start with some checkpoint. It should be fixed

File: rasa/core/training/generator.py (hunk 1 above)
Comment (Ghostvv) [reply rasa/core/training/generator.py:730]: but `md` format doesn't require this hack? I think there is smth wrong with yaml then

Comment (wochinge) [reply rasa/core/training/generator.py:730]: no, both require this hack.

**Comment by wochinge** [comment]
I'd simply disable the checkpoint warning for the `RulePolicy`, hm?

File: examples/rules/data/stories.md [hunk 2]
```diff
@@ -1,109 +0,0 @@
-<!-- each story starting with `>>` will be perceived as independent rule -->
```
Comment (Ghostvv) [inline examples/rules/data/stories.md:1]: do we still support `md` format? let's not remove that. E.g. for me it is much better format for stories

File: rasa/core/policies/fallback.py [hunk 3]
```diff
@@ -57,6 +58,13 @@ def __init__(
         self.core_threshold = core_threshold
         self.fallback_action_name = fallback_action_name
 
+        common_utils.raise_warning(
+            f"'{self.__class__.__name__}' is deprecated and will be removed in version "
+            "3.0. It is recommended to use the 'RulePolicy' instead.",
```
Comment (Ghostvv) [inline rasa/core/policies/fallback.py:63]: are we really going to drag old style up to version `3`?

Comment (wochinge) [reply rasa/core/policies/fallback.py:63]: We can also just say "in the future"?

File: examples/rules/data/stories.md (hunk 2 above)
Comment (wochinge) [reply examples/rules/data/stories.md:1]: time for change 😆 I can't put both, yaml and md, in there as then everything would bee duplicate. I'm also not sure if we should introduce new features to the old training data format.

File: rasa/core/training/generator.py (hunk 1 above)
Comment (Ghostvv) [reply rasa/core/training/generator.py]: wait, isn't this hack disables the warning for all non rule stories? that's not a good idea

File: examples/rules/data/stories.md (hunk 2 above)
Comment (Ghostvv) [reply examples/rules/data/stories.md:1]: we should

Comment (Ghostvv) [reply examples/rules/data/stories.md:1]: I'm planning to use md format

Comment (Ghostvv) [reply examples/rules/data/stories.md:1]: let's just put in a different folder `stories_md` or smth

File: rasa/core/policies/fallback.py (hunk 3 above)
Comment (Ghostvv) [reply rasa/core/policies/fallback.py:63]: yes, let's do that

**Pull request merged**